borgmatic/hooks/mysql.py

```diff
@@ -225,31 +236,69 @@ def restore_data_source_dump(
     mysql_restore_command = tuple(
         shlex.quote(part) for part in shlex.split(data_source.get('mysql_command') or 'mysql')
     )
-    restore_command = (
-        mysql_restore_command
-        + ('--batch',)
-        + (
-            tuple(data_source['restore_options'].split(' '))
-            if 'restore_options' in data_source
-            else ()
+
+    dump_format = data_source.get('format')
+    is_directory_format = dump_format == 'directory'
+
+    if is_directory_format:
+        mysql_restore_command = tuple(
+            shlex.quote(part) for part in shlex.split(data_source.get('mysql_command') or 'mysql')
         )
-        + (('--host', hostname) if hostname else ())
-        + (('--port', str(port)) if port else ())
-        + (('--protocol', 'tcp') if hostname or port else ())
-        + (('--user', username) if username else ())
-    )
+        restore_command = (
+            mysql_restore_command
+            + ('--batch',)
+            + (
+                tuple(data_source['restore_options'].split(' '))
+                if 'restore_options' in data_source
+                else ()
+            )
+            + (('--host', hostname) if hostname else ())
+            + (('--port', str(port)) if port else ())
+            + (('--protocol', 'tcp') if hostname or port else ())
+            + (('--user', username) if username else ())
+        )
+    else:
+        restore_command = (
+            mysql_restore_command
+            + ('--batch',)
+            + (
+                tuple(data_source['restore_options'].split(' '))
+                if 'restore_options' in data_source
+                else ()
+            )
+            + (('--host', hostname) if hostname else ())
+            + (('--port', str(port)) if port else ())
+            + (('--protocol', 'tcp') if hostname or port else ())
+            + (('--user', username) if username else ())
+        )
+
     extra_environment = {'MYSQL_PWD': password} if password else None
 
     logger.debug(f"{log_prefix}: Restoring MySQL database {data_source['name']}{dry_run_label}")
     if dry_run:
         return
 
-    # Don't give Borg local path so as to error on warnings, as "borg extract" only gives a warning
-    # if the restore paths don't exist in the archive.
-    execute_command_with_processes(
-        restore_command,
-        [extract_process],
-        output_log_level=logging.DEBUG,
-        input_file=extract_process.stdout,
-        extra_environment=extra_environment,
-    )
+    if is_directory_format:
+        dump_path = make_dump_path(config)
+        dump_directory = dump.make_data_source_dump_filename(
+            dump_path, data_source['name'], data_source.get('hostname')
+        )
+        for file in os.listdir(dump_directory):
+            file_path = os.path.join(dump_directory, file)
+            if file.endswith('.sql'):
```

Comment:
[The docs say]( https://dev.mysql.com/doc/refman/8.4/en/mysqldump.html#option_mysqldump_tab)these are `.txt` files...?